edb/server/dbview/dbview.pyx

```diff
@@ -837,15 +843,45 @@ cdef class DatabaseConnectionView:
     cdef cache_compiled_query(self, object key, object query_unit_group):
         assert query_unit_group.cacheable
 
-        self._db._cache_compiled_query(key, query_unit_group)
+        if self._in_tx:
+            if (
+                not self._tx_error and
+                self._query_cache_enabled and
+                not self._in_tx_with_ddl and
+                key not in self._in_tx_eql_to_compiled
+            ):
+                self._in_tx_eql_to_compiled[key] = query_unit_group
+                while self._in_tx_eql_to_compiled.needs_cleanup():
+                    self._in_tx_eql_to_compiled.cleanup_one()
+        else:
+            self._db._cache_compiled_query(key, query_unit_group)
```

Comment:
Is this the behavior we want?

For transactions without any DDL, is there anything stopping us from calling `_cache_compiled_query` immediately?

Reply:
Now that I'm reviewing this, perhaps we can still cache the entries compiled in transactions (without DDL for now). Let me try that with a new way to deal with the inline cache.